lightly/cli/_helpers.py

```diff
@@ -65,3 +65,65 @@ def load_state_dict_from_url(url, map_location=None):
     # in this case downloading the pre-trained model was not possible
     # notify the user and return
     return {'state_dict': None}
+
+
+def _expand_batchnorm_weights(model_dict, state_dict, num_splits):
+    """Expands the weights of the BatchNorm2d to the size of SplitBatchNorm.
+
+    """
+    running_mean = 'running_mean'
+    running_var = 'running_var'
+
+    for key, item in model_dict.items():
+        # not batchnorm -> continue
+        if not running_mean in key and not running_var in key:
+            continue
+        
+        state = state_dict.get(key, None)
+        # not in dict -> continue
+        if state is None:
+            continue
+        # same shape -> continue
+        if item.shape == state.shape:
+            continue
+
+        # found running mean or running var with different shapes
+        state_dict[key] = state.repeat(num_splits)
+
+    return state_dict
+
+
+def _filter_state_dict(state_dict):
+    """Prevents unexpected key error when loading PyTorch-Lightning checkpoints.
+
+    Removes the "model." prefix from all keys in the state dictionary.
+
+    """
+    # 
+    new_state_dict = {}
+    for key, item in state_dict.items():
+        key_parts = key.split('.')[1:]
+        key_parts = [k if k != 'features' else 'backbone' for k in key_parts]
+        new_key = '.'.join(key_parts)
+        new_state_dict[new_key] = item
+
+    return new_state_dict
+
+
+def load_from_state_dict(model,
+                         state_dict,
+                         strict: bool = True,
+                         apply_filter: bool = True):
+    """Loads the model weights from the state dictionary.
+
+    """
+    # step 1: filter state dict
+    if apply_filter:
+        state_dict = _filter_state_dict(state_dict)
+ 
+    # step 2: expand batchnorm weights
+    state_dict = \
+        _expand_batchnorm_weights(model.state_dict(), state_dict, 0)
```

Comment:
this feels like an initialization, would there be a way of using a closure or lightweight encapsulation to make this easier to read? Also if the function signature doesn't have 'splits' in it should probably pass 0 as a named variable at least

Reply:
not to nitpick too much about the function name but 'maybe' might be useful prefix if the transformation isn't guaranteed to occur (which appears it is not to be)